### ISSUE TITLE
Bug: Pointer over / out events do not fire correctly when using custom hitArea

### ISSUE THREAD

**Issue by Zyie** [pixijs/pixijs#9284]
### Current Behavior

pointer over/out events fire when the mouse moves over the stage when using a custom hitArea

### Expected Behavior

pointer over/out events fire over the bounds of the DisplayObject

### Steps to Reproduce

example: https://jsfiddle.net/0dwogvej/

```ts
const app = new PIXI.Application();
document.body.appendChild(app.view);

const sprites1 = new PIXI.Graphics().beginFill(0xffffff).drawRect(0,0,100,100)

app.stage.addChild(sprites1);
sprites1.interactive = true;

sprites1.hitArea = new PIXI.Rectangle(0, 0, 100, 100)
sprites1.on('pointerover', () => console.log('over'))
sprites1.on('pointerout', () => console.log('out'))
```

### Environment

- **`pixi.js` version**: *7.2.1+*
- **Running Example**: *https://jsfiddle.net/0dwogvej/*

### Additional Information

This bug was introduced with #9273 which aimed to fix global move events

**Comment by MuTsunTsai** [comment]
I'm experiencing the same issue. Also, I noticed that since v7.2.1+, pointer events are fired even at supposedly masked area of objects. Masks used to work as expected in v7.2.0 though.

**Comment by MuTsunTsai** [comment]
Confirm issue fixed in v7.2.3.
